cmd/kops/toolbox_instance_selector.go

```diff
@@ -0,0 +1,487 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/aws/amazon-ec2-instance-selector/pkg/cli"
+	"github.com/aws/amazon-ec2-instance-selector/pkg/selector"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kops/cmd/kops/util"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/client/simple"
+	"k8s.io/kops/upup/pkg/fi/cloudup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// Filter Flag Constants
+const (
+	vcpus                  = "vcpus"
```

Comment:
Out of curiosity, would it make sense to reference these constants from somewhere else rather than redefine them all here? I see similar constants [in the upstream library](https://github.com/aws/amazon-ec2-instance-selector/blob/master/pkg/selector/selector.go) but they're not exported, is that intentional?

Reply:
I'm not sure it makes sense to expose those constants. They are similar, but they're used to build up the internal filter hashmap. The constants here are meant to be user-friendly flag names.  We're looking at replacing those constants in the upstream anyways, to allow for more dynamically created filters.